### PR TITLE
Use pre-built cargo-hack in CI

### PR DESCRIPTION
**Pull request by LingMan**
Use pre-built cargo-hack in CI
    
The authors of cargo-hack provide an Action to use in CI and avoid having to build from source.
Currently the MSRV CI job fails because latest cargo-hack requires 1.56 to build while the test is
run with 1.46.
Simply running cargo-hack only requies 1.26 though.
    
This should also speedup CI runs, since we don't have to build cargo-hack each time.

### CODE REVIEW

**Comment by Thomasdezeeuw** [comment]
I've just started the v0.4.x branch so it's not actually used yet. So let's wait until we merge https://github.com/rust-lang/socket2/pull/320 and we actually start with the master/v0.4 split.

**Comment by LingMan** [comment]
Alright, guess you'll add the v0.4.x branch to CI with #320 then. I've dropped the second commit and updated the PR description.

**Review by Thomasdezeeuw — APPROVED** [review]

**Comment by Thomasdezeeuw** [comment]
Thanks @LingMan.

**Pull request merged**
